### PR TITLE
cmd/snap-confine: use /etc/ssl from the core snap

### PR DESCRIPTION
**Pull request by morphis**
We need to use /etc/ssl from the core snap next to /etc/alternatives
as applications like LXD expect a correct setup of the certificate
chain which is broken if we're running snaps on non Ubuntu/Debian
distributions.

Fixes https://bugzilla.opensuse.org/show_bug.cgi?id=1028568

### CODE REVIEW

File: tests/main/dirs-not-shared-with-host/task.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+summary: Ensure that certain directories are coming from the core snap
```
Comment (zyga) [inline tests/main/dirs-not-shared-with-host/task.yaml]: Nice, thanks!

File: tests/main/dirs-not-shared-with-host/task.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+description: |
```
Comment (zyga) [inline tests/main/dirs-not-shared-with-host/task.yaml]: I think you meant:
```
details: |
  The snap-confine program bind mounts the /etc directory from the classic distribution into the snap execution
  environment. Certain directories however, if they exist on the host's /etc are actually, bind-mounted from the
  core snap for a more consistent behaviour across various distributions.
```

File: cmd/snap-confine/mount-support.c [hunk 3]
```diff
@@ -341,20 +341,26 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
...
-				 etc_alternatives);
-		sc_do_mount(src, dst, NULL, MS_BIND, NULL);
-		sc_do_mount("none", dst, NULL, MS_SLAVE, NULL);
+	if (config->on_classic_distro) {
+		// Since we mounted /etc from the host filesystem to the scratch directory,
+		// we may need to put certain directories from the desired root filesystem
+		// (e.g. the core snap) back. This way the behavior of running snaps is not
+		// affected by the alternatives directory from the host, if one exists.
+		//
+		// https://bugs.launchpad.net/snap-confine/+bug/1580018
```
Comment (chipaca) [inline cmd/snap-confine/mount-support.c]: could you expand this comment a little bit to also link the bug this PR fixes?

Comment (morphis) [inline cmd/snap-confine/mount-support.c]: Done.

**Comment by jdstrand** [comment]
This PR is a 6 of one half dozen of the other type of thing. This PR means that now locally installed certificates on classic won't work and there is no mechanism for extending the core snap's certificates. Debian and Ubuntu have 'update-ca-certificates' (`man 8 update-ca-certificates`) which can look at /usr/local/share/ca-certificates that might be used to help address this, but I don't know what mechanisms are available in other distros. I suspect more people are affected by this PR not being applied than those that will be when it is applied, but I predict bug reports.

**Comment by anlx-sw** [comment]
is there a way to persistently add root certificates to snap? i run snap on a ubuntu 16.04 server and it seems that tls certificates from an internal ca can't be verified anymore.

**Pull request merged**